pkg/verify/signature.go

```diff
@@ -234,3 +242,29 @@ func verifyMessageSignatureWithArtifactDigest(verifier signature.Verifier, msg M
 
 	return nil
 }
+
+func GetHashForDigestAlgorithm(digestAlgorithm string) (crypto.Hash, error) {
+	var hashFunc crypto.Hash
+	switch digestAlgorithm {
+	case "sha256":
+		hashFunc = crypto.SHA256
+	case "sha384":
+		hashFunc = crypto.SHA384
+	case "sha512":
+		hashFunc = crypto.SHA512
+	default:
+		return crypto.Hash(0), fmt.Errorf("unrecognized digest algorithm: %s", digestAlgorithm)
+	}
+	return hashFunc, nil
+}
+
+func getHashForHashAlgorithmString(hashAlgorithm string) (crypto.Hash, error) {
```

Comment:
This is the string representation of the `HashAlgorithm` enum which gets used in the case where we're providing a trusted root (rather than public key material). Currently, different hash algorithms only work in the case where we provide trusted public key material since this enum only has a `SHA2_256` value. But if we add more values to this enum, we should be able to "unlock" different hash algorithms.